src/vs/workbench/contrib/inlineChat/browser/inlineChatController.ts

```diff
@@ -185,7 +185,16 @@ export class InlineChatController implements IEditorContribution {
 				}
 			}
 
-			return this._store.add(_instaService.createInstance(InlineChatZoneWidget, location, this._editor));
+			const zone = _instaService.createInstance(InlineChatZoneWidget, location, this._editor);
+			this._store.add(zone);
+			this._store.add(zone.widget.chatWidget.onDidClear(async () => {
+				const r = this.joinCurrentRun();
+				this.cancelSession();
+				await r;
```

Comment:
This might cause problems if something triggers `onDidClear` in quick succession (maybe the user double clicking some refresh button?), before `r` resolves.